redfish-core/lib/pcie_slots.hpp

```diff
@@ -29,6 +31,55 @@
 namespace redfish
 {
 
+inline void
+    addLinkedPcieDevices(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& slotPath, size_t index)
+{
+    // Collect device associated with this slot and
+    // populate it here
+    constexpr std::array<std::string_view, 1> interfaces = {
+        "xyz.openbmc_project.Inventory.Item.PCIeDevice"};
+    dbus::utility::getSubTree(
+        slotPath, 0, interfaces,
+        [asyncResp,
+         index](const boost::system::error_code& ec,
+                const dbus::utility::MapperGetSubTreeResponse& subtree) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("D-Bus response error on GetSubTree {}",
+                             ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        if (subtree.empty())
+        {
+            BMCWEB_LOG_DEBUG(
+                "Can't find PCIeDevice D-Bus object for given slot");
+            return;
+        }
+
+        // Assuming only one device path per slot.
+        const std::string& pcieDevciePath = std::get<0>(subtree[0]);
```

Comment:
nit: typo on variable name here. See it was same as 1060, so not sure if worth changing.

"Devcie" should be "Device"

Reply:
Changed